modules/diskq/qdisk.c

```diff
@@ -519,39 +519,38 @@ _read_record_from_disk(QDisk *self, GString *record, guint32 record_length)
   return TRUE;
 }
 
-static inline gint64
-_calculate_new_read_head_position(QDisk *self, guint32 record_length)
+static inline void
+_maybe_apply_non_reliable_corrections(QDisk *self)
 {
-  gint64 new_read_head_position = self->hdr->read_head + record_length + sizeof(record_length);
-
-  if (new_read_head_position > self->hdr->write_head)
-    new_read_head_position = _correct_position_if_max_size_is_reached(self, &new_read_head_position);
+  if (self->options->reliable)
+    return;
 
-  return new_read_head_position;
+  qdisk_empty_backlog(self);
+  g_assert(self->hdr->backlog_len == 0);
```

Comment:
Optional: I think, the assert is unnecessary here. 
From the commit history, I get why is here, but for the end "production" I don't think is needed here.

Reply:
This was already here, I just did not notice this should be removed.
It has become a tautology.

Thanks, I'll remove it.